pkg/maps/ctmap/gc/gc.go

```diff
@@ -51,18 +51,47 @@ func Enable(ipv4, ipv6 bool, restoredEndpoints []*endpoint.Endpoint, mgr Endpoin
 		ipv4Orig := ipv4
 		ipv6Orig := ipv6
 		for {
-			var maxDeleteRatio float64
+			var (
+				maxDeleteRatio float64
+
+				// gcStart and matchCB are used to populate DNSPendingDelete fields on
+				// endpoints. These hold IPs that are deletable in the DNS caches, but
+				// may be in use by connections. Each loop of this GC keeps those
+				// entries alive by touching them in matchCB. We also need to record
+				// the start of each CT GC loop (further below in the goroutine). In
+				// all cases the timestamp used is the start of the GC loop. This
+				// simplifies the logic to determine if a marked connection was marked
+				// in the most recent GC loop or not: if the active timestamp is before
+				// the recorded start of the GC loop then it must mean the next
+				// iteration has completed and it is not in-use.
+				gcStart = time.Now()
+				matchCB = func(srcIP net.IP, dstIP net.IP, dstPort uint16, nextHdr, flags uint8, entry *ctmap.CtEntry) bool {
+					// Ensure we only look at outbound connections that can be FQDN related
+					if flags != ctmap.TUPLE_F_OUT {
+						return true
+					}
+					if ep := mgr.LookupIP(srcIP); ep != nil {
```

Comment:
This lookup will get an internal mutex for the manager for each CT entry. Can't we simply do a `mgr.GetEndpoints` before iterating CT map and do a lockless lookup map instead? We already have all endpoints on line `80` we just need to transform into a `map[ipStr]*Endpoint`

Reply:
I actually had it that way at the start of this thing... At the time it didn't make sense to keep it, but the way it works now it will be fine. I switched it back